spock-specs/src/test/groovy/org/spockframework/smoke/FinalFields.groovy

```diff
@@ -0,0 +1,102 @@
+package org.spockframework.smoke
+
+import org.spockframework.EmbeddedSpecification
+import org.spockframework.compiler.InvalidSpecCompileException
+
+import spock.lang.Shared
+import spock.lang.Specification
+
+class FinalFields extends Specification {
+    @Shared
+    final String finalShared = 'final shared field'
+
+    @Shared
+    String normalShared = 'normal shared field'
+
+    final String finalField = 'final field'
+
+    String normalField = 'normal field'
+
+    def "can modify normal fields"() {
+        expect:
+        normalField == 'normal field'
+
+        when:
+        normalField = 'new value'
+
+        then:
+        normalField == 'new value'
+    }
+
+    def "can modify normal shared fields"() {
+        expect:
+        normalShared == 'normal shared field'
+
+        when:
+        normalShared = 'new value'
+
+        then:
+        normalShared == 'new value'
+    }
+
+    def "cannot modify final fields"() {
+        expect:
+        finalField == 'final field'
+
+        when:
+        finalField = 'new value'
+
+        then:
+        ReadOnlyPropertyException ex = thrown()
+        ex.message == 'Cannot set readonly property: finalField for class: org.spockframework.smoke.FinalFields'
+        finalField == 'final field'
+    }
+
+    def "cannot modify final shared fields"() {
+        expect:
+        finalShared == 'final shared field'
+
+        when:
+        finalShared = 'new value'
+
+        then:
+        ReadOnlyPropertyException ex = thrown()
+        ex.message == 'Cannot set readonly property: finalShared for class: org.spockframework.smoke.FinalFields'
+    }
```

Comment:
In wonder, why you omitted checking if the shared value wasn't in fact set (as you do in the test before)?